### PR TITLE
Fix von mises log prob and sample

### PR DESCRIPTION
**Pull request by fehiepsi**
Hopefully, this will be helpful for #774.

This fixes VonMises.log_prob and `von_mises_centered` sampler. They are tested against scipy. In the master branch, we missed taking `log` after applying Bessel function on concentration parameter. In addition, because `bessel_i0e` is exponentially scaled Bessel function, we need to plus `concentration` after taking the log. In addition, I made some enhancements:
+ using `jax.scipy.special.i1e` instead of `lax.bessel_i1e` because the latter can't be applied on int scalar/arrays.
+ scale `value - loc` to [0, 2pi] before taking `cos`. This is helpful when `value - loc` is very large, which causes discontinuity as in the picture below
image

Blocked by #777

cc @Nikolin4 @OlaRonning

### CODE REVIEW

**Review by Nikolin4 — APPROVED** [review]

**Review by Nikolin4 — APPROVED** [review]

**Comment by fehiepsi** [comment]
@OlaRonning Could you help me review this PR? I think those fixes would be helpful for your experiments with VonMises.

**Comment by OlaRonning** [comment]
Hi @fehiepsi, I'll have a look this afternoon.

File: numpyro/distributions/util.py [hunk 1]
```diff
@@ -420,7 +420,7 @@ def body_fn(*args):
         w = jnp.where(done, w, (1. + s * z) / (s + z))  # Update where not done
 
         y = concentration * (s - w)
-        v = random.uniform(key=uni_vkey, shape=shape, dtype=concentration.dtype, minval=-1., maxval=1.)
+        v = random.uniform(key=uni_vkey, shape=shape, dtype=concentration.dtype)
```
Comment (OlaRonning) [inline numpyro/distributions/util.py:423]: Is the reasoning that we can remove the bounds that [-1,1] ~ Reals?

**Review by OlaRonning — APPROVED** [review]
See question below, otherwise LGTM, well-done @fehiepsi!

**Review by neerajprad — APPROVED** [review]

**Pull request merged**